common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/YubiKitCertBasedAuthManager.java

```diff
@@ -0,0 +1,220 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers;
+
+import android.app.Activity;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.microsoft.identity.common.internal.telemetry.Telemetry;
+import com.microsoft.identity.common.java.telemetry.events.PivProviderStatusEvent;
+import com.microsoft.identity.common.logging.Logger;
+import com.yubico.yubikit.android.YubiKitManager;
+import com.yubico.yubikit.android.transport.usb.UsbConfiguration;
+import com.yubico.yubikit.android.transport.usb.UsbYubiKeyDevice;
+import com.yubico.yubikit.android.transport.usb.connection.UsbSmartCardConnection;
+import com.yubico.yubikit.core.smartcard.SmartCardConnection;
+import com.yubico.yubikit.core.util.Callback;
+import com.yubico.yubikit.core.util.Result;
+import com.yubico.yubikit.piv.PivSession;
+import com.yubico.yubikit.piv.jca.PivProvider;
+
+import java.io.IOException;
+import java.security.Security;
+import java.util.concurrent.Callable;
+
+/**
+ * Utilizes YubiKit in order to detect and interact with YubiKeys for smartcard certificate based authentication.
+ */
+public class YubiKitCertBasedAuthManager implements ISmartcardCertBasedAuthManager {
+
+    private static final String TAG = YubiKitCertBasedAuthManager.class.getSimpleName();
+    private static final String MDEVICE_NULL_ERROR_MESSAGE = "Instance UsbYubiKitDevice variable (mDevice) is null.";
+    private static final String YUBIKEY_PROVIDER = "YKPiv";
+
+    private final YubiKitManager mYubiKitManager;
+    private UsbYubiKeyDevice mDevice;
+    //Lock to help facilitate synchronization
+    private static final Object sDeviceLock = new Object();
+
+    /**
+     * Create new instance of YubiKitCertBasedAuthManager.
+     * @param activity current host activity.
+     */
+    YubiKitCertBasedAuthManager(Activity activity) {
+        mYubiKitManager = new YubiKitManager(activity.getApplicationContext());
```

Comment:
👀

For principle of least privilege, consider accepting the least-specific class you need to fulfill this class' requirements. Looks like what you really is a `Context`, rather than the whole `Activity`